### PR TITLE
build: use golang 1.20

### PR DESCRIPTION
**Pull request by maiqueb**
**What this PR does / why we need it**:
Bump to golang 1.20

**Special notes for your reviewer**:

**Release note**:

```release-note
Bump to golang 1.20
```

### CODE REVIEW

File: automation/check-patch.setup.sh [hunk 1]
```diff
@@ -12,8 +12,8 @@ export PATH=${GOPATH}/bin:${GOROOT}/bin:${PATH}
 export GOBIN=${GOROOT}/bin/
 mkdir -p $GOBIN
 
-echo 'Install Go 1.19'
-export GIMME_GO_VERSION=1.19
+echo 'Install Go 1.20'
+export GIMME_GO_VERSION=1.20
```
Comment (oshoval) [inline automation/check-patch.setup.sh:16]: if you want a followup you can do as we are doing on CNAO
auto read the version from go.mod using a script
and then one less place to update
(maybe same trick can be done with ARG for dockerfile)

**Review by oshoval — APPROVED** [review]

File: automation/check-patch.setup.sh (hunk 1 above)
Comment (maiqueb) [reply automation/check-patch.setup.sh:16]: That would be nice to have.

**Pull request merged**